test/e2e/reconciler/ingress/controller_test.go

```diff
@@ -93,9 +94,14 @@ func TestIngressController(t *testing.T) {
 						return false
 					}
 					if len(got.Items) != 1 {
+						klog.Errorf("expected 1 ingress, got %d", len(got.Items))
```

Comment:
can we list the splitted ingresses too? Are those 1. existing and 2. scheduled ?

Reply:
In general, we shouldn't have these giant e2e tests with steps like 1. do X 2. magic happens with a huge machinery 🪄 3. some outcome a million lines of code later.

Reply:
I think golden-path e2e's are fine - they're as much about validating behavior from a user perspective as anything else. But to your point, they're only as good as the lower levels of testing that provide detailed coverage for all the corner cases.